crates/accelerate/src/unitary_synthesis.rs

```diff
@@ -881,6 +968,7 @@ fn preferred_direction(
 }
 
 fn synth_su4_sequence(
+    py: Python,
```

Comment:
is there a way to avoid using `py` here?

Reply:
If you need it to instantiate the decomposer, I'm afraid there's no way to avoid it.

Reply:
I managed to remove the need of py in [231af7f](https://github.com/Qiskit/qiskit/pull/13568/commits/231af7f18b4ce6b69d82c526d1c3d2488c899f91)
by replacing it with gil, as suggested by Matthew in https://github.com/Qiskit/qiskit/pull/13419.